mail_tracking/models/mail_tracking_email.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import urlparse
+import time
+from datetime import datetime
+
+from openerp import models, api, fields, tools
+import openerp.addons.decimal_precision as dp
+
+_logger = logging.getLogger(__name__)
+
+
+class MailTrackingEmail(models.Model):
+    _name = "mail.tracking.email"
+    _order = 'time desc'
+    _rec_name = 'name'
+    _description = 'MailTracking email'
+
+    name = fields.Char(string="Subject", readonly=True)
+    timestamp = fields.Float(
+        string='UTC timestamp', readonly=True,
+        digits=dp.get_precision('MailTracking Timestamp'))
+    time = fields.Datetime(string="Time", readonly=True)
+    date = fields.Date(string="Date", readonly=True)
+    mail_message_id = fields.Many2one(
+        string="Message", comodel_name='mail.message', readonly=True)
+    mail_id = fields.Many2one(
+        string="Email", comodel_name='mail.mail', readonly=True)
+    partner_id = fields.Many2one(
+        string="Partner", comodel_name='res.partner', readonly=True)
+    recipient = fields.Char(string='Recipient email', readonly=True)
+    sender = fields.Char(string='Sender email', readonly=True)
+    state = fields.Selection([
```

Comment:
I may be misunderstanding how this is working, because I don't see reading of the SMTP Diagnostic Code header in order to determine delivery status, but if we are actually inspecting this header it would also be nice to store the status codes header. 

As a mail admin, these are more granular and could help in troubleshooting outbound issues. There are a lot of RFCs for this, and all may not be implemented on each server, but they operate in a way similar to HTTP status codes. A fairly complete list can be seen here - http://www.iana.org/assignments/smtp-enhanced-status-codes/smtp-enhanced-status-codes.xhtml